### DISCUSSION TITLE
Opentelemetry support

### DISCUSSION THREAD

**Discussion by weyert** [opstrace/opstrace#569] [Q&A]
I was wondering if I could use Opentelemetry to send metrics and traces to opstrace using the OTLP format if possible? I didn’t find much reference to it. My current projects are all instrumented with Opentelemetry

**Comment by jgehrcke** [comment]
Hey!

>  OTLP format 

For reference and context, and anyone wondering:
* The protocol aspects are specified here: https://github.com/open-telemetry/opentelemetry-specification/blob/0a7781cb12cc24797f22538ad71665b8bfdc5efe/specification/protocol/otlp.md
* The binary message (protobuf) formats are specified here: https://github.com/open-telemetry/opentelemetry-proto/tree/493f9f8f3bebd6c9019df77723819522fcd0873e/opentelemetry/proto

@weyert as of today, Opstrace does not expose OTLP-compatible interfaces. Not for logs, not for metrics, not for traces. Also note that for traces, Opstrace does not have a storage engine yet.

In the future, I am sure that Opstrace will provide these interfaces -- this just makes a lot of sense! OTLP has a lot of momentum. So, your question is great and an obvious one. At this point we can't make definite statements about what comes when; but let's be optimistic.

Ad documentation: we will have to first-class the "data API" aspect of Opstrace in the docs:
- make it easily discoverable from the side bar
- explain exactly which API endpoints are exposed; linking to external reference docs where appropriate

Here is what's there to date: https://opstrace.com/docs/references/concepts#data-api
